XenModel/XenAPI-Extensions/Pool.cs

```diff
@@ -552,13 +560,30 @@ public static int GetDefaultTime()
         }
         #endregion
 
-        public string GetUploadToken(IXenConnection connection)
+        public string GetSecretyInfo(IXenConnection connection, string secretType)
         {
-            if (connection == null || string.IsNullOrEmpty(UploadTokenSecretUuid))
+            string UUID = string.Empty;
+            switch (secretType)
+            {
+                case CallHomeSettings.UPLOAD_CREDENTIAL_USER_SECRET:
+                    UUID = PasswordSecretUuid;

```

Comment:
I think this should be: UUID = UserNameSecretUuid;
And three lined below:  UUID = PasswordSecretUuid;